### ISSUE TITLE
Unable to expand/edit codings added via the interface

### ISSUE THREAD

**Issue by broadwell** [sul-cidr/sati#24]
To reproduce: create an item, add a coding, save the item, browse to the item's codings, try to expand a coding (to view or edit it) by clicking on the arrow or the label (e.g., `Coding:
Coding by broadwell@stanford.edu, 2021-01-14`).

It seems as though the click event isn't triggering the intended style change to expand the coding display element.

This has only been tested/reproduced with dummy test items created from scratch via the interface. Not sure if it would happen with codings added to items created via bulk update.

**Comment by simonwiles** [comment]
I'm not able to replicate this yet -- I assume there are no JS errors in the console?

Edit: I take that back -- I can replicate on FF only.  Will investigate.

Edit edit: Nope, my mistake -- it's a race condition with the signal that invokes function that adds the event handles (sometimes) firing before the relevant part of the DOM is ready.  Fix incoming.  Thanks :)

**Comment by broadwell** [comment]
As I mentioned in the comment on #25, this is still problematic on Mac OS FF (84.0.2 (64-bit)). Something interesting I just noticed, though: if I reload an item's codings change/list page (e.g., http://localhost:8000/admin/items/item/72f79922-0f41-482a-9536-5df4ee2723b2/change/#inline-itemcoding), then the expand functionality works. But if I navigate to an item via the Items list and then click the "Codings" tab, then the expand behavior doesn't work (unless I reload the page). So probably the race condition is still lurking in Firefox land?

**Comment by simonwiles** [comment]
So I dug into the django-baton source code, and I tried `promise`-ifying the dispatch code so I could wait for both the `onReady` and `onTabsReady` events before attempting to add the event listeners, and nothing I did eliminated the problem completely.  So I've just added a `MutationObserver` which fires the code which adds the event handlers when the elements appear in the DOM, if they're not already there.  Little bit of a sledge-hammer / nut situation, but hopefully it should cover all eventualities.
